sunscreen_zkp_backend/src/lib.rs

```diff
@@ -126,7 +127,17 @@ pub trait Gadget: Any + Send + Sync {
     }
 }
 
+impl Serialize for dyn Gadget {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.debug_name())
```

Comment:
Hmm... I don't approve of this. The expectation when you serialize something is that you can deserialize it and get a the exact thing back. This cannot be the case with Gadgets because they contain code.

Unfortunately, there isn't a simple alternative; you need to implement serialization that isn't serialization.